File: NapakalakiRuby/lib/napakalaki.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

require 'singleton'

class Napakalaki
  include Singleton
  
  public
  
  def initialize
    @currentPlayerIndex
    @currentMonster
    @currentPlayer
    @players
    @dealer
  end

  private
  def initPlayers(names)
    #No se sabe
  end
  
  def nextPlayer
    #No se sabe
  end
  
  def nextTurnAllowed
    #No se sabe
  end
  
  def setEnemies
    #No se sabe
  end
  
  public
  
  def developCombat
    #No se sabe
  end
  
  def discardVisibleTreasures(treasures)
    #No se sabe
  end
  
  def discardHiddenTreasures(treasures)
    #No se sabe
  end
  
  def makeTreasuresVisible(treasures)
    #No se sabe
  end
  
  def initGame(players)
    
  end
  
  def getCurrentPlayer
    #No se sabe
  end
  
  def getCurrentMonster
    #No se sabe
  end
  
  def nextTurn
    #No se sabe
  end
  
  def endOfGame(result)
    #No se sabe
  end 
end

end

File: NapakalakiJava/src/NapakalakiGame/Napakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.*;
/**
 *
 * @author jomabose
 */
public class Napakalaki {
    
    private static final Napakalaki instance = new Napakalaki();
    
    private Monster currentMonster;
    private CardDealer dealer;
    private Player currentPlayer;
    private ArrayList<Player> players;
    
    private Napakalaki() {
        //currentMonster
        //dealer
        //currentPlayer
        players = new ArrayList();
    }
    private void initPlayers(String[] name){
        //No se sabe
    }
    private Player nextPlayer(){
        //No se sabe
    }
    private boolean nextTurnAllowed(){
        //No se sabe
    }
    private void setEnemies(){
        //No se sabe
    }
    
    public static Napakalaki getInstance() {
        return instance;
    }
    public CombatResult developCombat(){
        //No se sabe
    }
    public void discardVisibleTreasures(Treasure[] treasures){
        //No se sabe
    }
    public void discardHiddenTreasures(Treasure[] treasures){
        //No se sabe
    }
    public void makeTreasuresVisible(Treasure[] treasures){
        //No se sabe
    }
    public void initGame(String[] players){
        //No se sabe
    }
    public Player getCurrentPlayer(){
        //No se sabe
    }
    public Monster getCurrentMonster(){
        //No se sabe
    }
    public boolean nextTurn(){
        //No se sabe
    }
    public boolean endOfGame(CombatResult result){
        //No se sabe
    }
    public String toString(){
        String cadena = "Monstruo actual: " + currentMonster.toString();
        cadena += "\nJugador actual: " + currentPlayer.toString();
        
        cadena += "\nTodos los jugadores: ";
        for(int i=0; i<players.size(); i++)
            cadena += "\n" + players.get(i).toString();
        cadena += "\nBaraja: " + dealer.toString();
        
        return cadena;
    }
}
